test/date_types.test.js

```diff
@@ -95,7 +97,19 @@ describe('MySQL DATE, DATETTIME, TIMESTAMP types on server with local TZ', funct
       },
     }, function(err, found) {
       assert.ok(!err);
-      assert.ok(found);
+      assert.equal(found.id, 1);
+      done();
+    });
+  });
+
+  it('should find model instance by date field passing string', function(done) {
```

Comment:
All "where" parameters are passed normalize function of "juggler". 
https://github.com/strongloop/loopback-datasource-juggler/blob/master/lib/dao.js#L1146
If property is one of type Date it does new Date(arg)
https://github.com/strongloop/loopback-datasource-juggler/blob/master/lib/dao.js#L1540

for DATE type test case will pass for timezones like my one (+xxxx):
"2016-12-22" -> "Thu Dec 22 2016 02:00:00 GMT+0200 (EET)"
but 
will fail for (-xxxx) timezones:
"2016-12-22" -> "Thu Dec 21 2016 22:00:00 GMT-0200"

I'm going to remove this test case but I'd suggest to mean it in documentation.